frontend/src/devicelist/__tests__/DeviceListPageContents.test.jsx

```diff
@@ -0,0 +1,196 @@
+      expect(wrapper.state().selected).toStrictEqual(expectedValue);
+    });
+  });
+  describe("handleSendersChange()", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<DeviceListPageContents />);
+    });
+    it("should set the state", () => {
+      const startingState = {
+        selected: 0,
+        senders: [],
+        receivers: []
+      };
+
+      const expectedValue = [new DeviceInfo("thing!")];
+
+      wrapper.setState(startingState);
+
+      wrapper.instance().handleSendersChange(expectedValue);
+      expect(wrapper.state().senders).toStrictEqual(expectedValue);
+    });
+  });
+  describe("handleReceiversChange()", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<DeviceListPageContents />);
+    });
+    it("should set the state", () => {
+      const startingState = {
+        selected: 0,
+        senders: [],
+        receivers: []
+      };
+
+      const expectedValue = [new DeviceInfo("thing!")];
+
+      wrapper.setState(startingState);
+
+      wrapper.instance().handleReceiversChange(expectedValue);
+      expect(wrapper.state().receivers).toStrictEqual(expectedValue);
+    });
+  });
+  describe("getDevices() function", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<DeviceListPageContents />);
+    });
+    it("if state.selected == 0, returns and array that combines state.senders and state.receivers ", () => {
+      const someState = {
+        selected: 0,
+        senders: dummySenders,
+        receivers: dummyReceivers
+      };
+      wrapper.setState(someState);
+      const result = wrapper.instance().getDevices();
+
+      expect(result).toStrictEqual(
+        someState.senders.concat(someState.receivers)
+      );
+    });
+    it("if state.selected == 1, returns state.senders ", () => {
+      const someState = {
+        selected: 1,
+        senders: dummySenders,
+        receivers: dummyReceivers
+      };
+      wrapper.setState(someState);
+      const result = wrapper.instance().getDevices();
+
+      expect(result).toStrictEqual(someState.senders);
+    });
+    it("if state.selected == 2, returns state.receivers ", () => {
+      const someState = {
+        selected: 2,
+        senders: dummySenders,
+        receivers: dummyReceivers
+      };
+      wrapper.setState(someState);
+      const result = wrapper.instance().getDevices();
+
+      expect(result).toStrictEqual(someState.receivers);
+    });
+  });
+  describe("getTitle() function", () => {
+    it("returns a <DeviceTableTitle/> component with expected props", () => {
+      wrapper = Enzyme.shallow(<DeviceListPageContents />);
+      const expected = {
+        index: wrapper.state().selected,
+        handleChange: wrapper.instance().handleChange
+      };
+      const wrap = Enzyme.shallow(wrapper.instance().getTitle());
+
+      //   expect(wrapper.matchesElement(<DeviceTableTitle index={expected.index} handleChange={expected.handleChange}/>)).toBe(true);
```

Comment:
Try this in order to also validate the type of the title
```suggestion
      const deviceTableTitle = wrapper.instance().getTitle();
      expect(deviceTableTitle.type).toEqual(DeviceTableTitle);
      
      const wrap = Enzyme.shallow(deviceTableTitle);
```